src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts

```diff
@@ -39,6 +47,24 @@ export function useEstimatedGasFee({
     const amount = gasSettings.isEIP1559 ? add(gasSettings.maxBaseFee, gasSettings.maxPriorityFee) : gasSettings.gasPrice;
 
     const totalWei = multiply(gasLimit, amount);
+
+    internalSelectedInputAsset.modify(asset => {
+      'worklet';
+
+      if (asset?.isNativeAsset) {
+        const gasFeeNativeCurrency = divWorklet(totalWei, 10 ** asset.decimals);
```

Comment:
leaving myself a note so I don't forget - not caused by this code but something I observe: sometimes I can get `gasLimit` back as `NaN` which messes up this `divWorklet`